src/extension.ts

```diff
@@ -1,51 +1,90 @@
-// The module 'vscode' contains the VS Code extensibility API
-// Import the module and reference it with the alias vscode in your code below
+import { exec, ExecException, execSync } from "child_process";
+import * as fs from "fs";
+import * as path from "path";
 import * as vscode from "vscode";
+import { getConfig } from "./config";
 import { preRunChecks } from "./preRunChecks";
 
+let extensionId: string;
```

Comment:
why is the value stored up here?
I think it would be better to avoide global scope. Please pass it to the function `adjustCurrentComponent` instead of saving it globally.

Reply:
Will be very likely that we will need this information again at other locations for other configurations. And it needs the context information from the `activate` method. But for now I moved it next to the `adjustCurrentComponent` function.